components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,294 @@
+                             const HorzMesh *Mesh,
+                             int NVertLevels,
+                             Config *Options);
+```
+
+#### 4.2.2 Initialization
+
+The init method will create the default vertical coordinate and return an error code:
+```c++
+int VertCoord::init();
+```
+
+#### 4.2.3 Retrieval
+
+There will be methods for getting the default and non-default vertical coordinate instances:
+```c++
+VertCoord *VertCoord::getDefault();
+VertCoord *VertCoord::get(const std::string &Name);
+```
+
+#### 4.2.4 Computation
+
+The public `computePressure` will sum the mass thicknesses times $g$ from the top layer down, starting with the surface pressure.
+This will be done with a `parallel_scan` inside a `parallel_for` over the mesh cells using hierarchical parallelism.
+```c++
+void VertCoord::computePressure(const Array2DReal &PressureInterface,
+                                const Array2DReal &PressureMid,
+                                const Array2DReal &LayerThickness,
+                                const Array2DReal &SurfacePressure) {
+
+}                                
+```
+
+The public `computeZHeight` will sum the mass thicknesses times $\alpha$ from the bottom later up, starting with the bottom elevation
+This will be done with a `parallel_scan` inside a `parallal_for over the mesh cells using hierarchical parallelism.
+```c++
+void VertCoord::computeZHeight(const Array2DReal &ZInterface,
+                               const Array2DReal &ZMid,
+                               const Array2DReal &LayerThickness,
+                               const Array2DReal &SpecVol,
+                               const Array2DReal &BottomDepth) {
+
+}    
+```
+
+The public `computeGeopotential` will sum together the $z$ height times $g$, the tidal potential, and self attraction and loading:
+```c++
+void VertCoord::computeGeopotential(const Array2DReal &GeopotentialMid,
+                                    const Array2DReal &ZMid,
+                                    const Array2DReal &TidalPotential
+                                    const Array2DReal &SelfAttractionLoading) {
+                                  
+} 
+```
+Tidal potential forcing and self attraction and loading will be default-off features.
+The will be added to (or excluded from) the geopotential based on config flags.
+
+The public `computePStarThickness` will determine the desired mass thickness used for the $p^\star$ vertical coordinate:
+```c++
+void VertCoord::computePStarThickness(const Array2DReal &LayerThicknessPStar,
+                                      const Array2DReal &VertCoordMovementWeights,
+                                      const Array2DReal &RefLayerThickness) {
+                                  
+} 
+```
+
+The private `minMaxLevelEdge` will determine the various vertical loop bounds on edges:
+```c++
+void VertCoord::minMaxLevelEdge(const Array2DI4 &MinLevelEdgeTop,
```

Comment:
Maybe we can just combine these into one called `minMaxLevel` that provides all of them? Unless there is an advantage to breaking it up?

Reply:
I think that makes sense.